### PR TITLE
Use Kupo instead of Ogmios for `utxosAt` query

### PR DESCRIPTION
**Pull request by errfrom**
Closes https://github.com/Plutonomicon/cardano-transaction-lib/issues/1184, Closes https://github.com/Plutonomicon/cardano-transaction-lib/issues/536, Closes https://github.com/Plutonomicon/cardano-transaction-lib/issues/1164, Closes https://github.com/Plutonomicon/cardano-transaction-lib/issues/1189

### Pre-review checklist

- [x] All code has been formatted using our config (`make format`)
- [x] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: nix/runtime.nix [hunk 1]
```diff
@@ -52,6 +52,16 @@ rec {
...
     };
+    kup = {
+      port = 1442;
+      # Do we actually need to sync the entire history?
+      since = "origin";
+      # This matches any Cardano address, "*/*" matches shelley addresses only, maybe we want the latter
+      match = "*";
+      #Do we want to pin to a concrete version or just "latest"?
+      tag = "v2.1.0";
+      # TODO: Do we want to support connection through ogmios?
```
Comment (jy14898) [inline nix/runtime.nix:63]: Would connecting via ogmios solve issues with synchronization issues between submission + utxos at?

File: nix/runtime.nix [hunk 2]
```diff
@@ -105,6 +115,7 @@ rec {
         );
       nodeDbVol = "node-${config.network.name}-db";
       nodeIpcVol = "node-${config.network.name}-ipc";
+      kupoDbVol = "kupo-db";
```
Comment (jy14898) [inline nix/runtime.nix:118]: Should we make this per-network?

**Review by klntsky — COMMENTED** [review]
Looks good, except of what Joseph has already pointed out

File: nix/runtime.nix (hunk 1 above)
Comment (errfrom) [reply nix/runtime.nix:60]: Created an issue: https://github.com/Plutonomicon/cardano-transaction-lib/issues/1226

File: webpack.config.js [hunk 3]
```diff
@@ -23,6 +23,11 @@ module.exports = {
 
   devServer: {
     port: 4008,
+    proxy: {
+      "/matches": process.env.KUPO_HOST, // e.g. http://localhost:1442
+      "/datums": process.env.KUPO_HOST,
+      "/scripts": process.env.KUPO_HOST,
+    },
```
[the remark this answers is not in the document]
Comment (errfrom) [reply webpack.config.js:30]: Created an issue: https://github.com/Plutonomicon/cardano-transaction-lib/issues/1227

**Review by klntsky — CHANGES_REQUESTED** [review]

File: templates/ctl-scaffold/packages.dhall [hunk 4]
```diff
@@ -367,7 +367,7 @@ let additions =
           , "variant"
           ]
         , repo = "https://github.com/Plutonomicon/cardano-transaction-lib.git"
-        , version = "b56e8d41d504d8db4a64b6b655281bc278f534b3"
+        , version = "b8c20088aa942281200b6d9e48baa833f3efc950"
```
Comment (klntsky) [inline templates/ctl-scaffold/packages.dhall:370]: Looks like you forgot to update `spago-packages.nix` here. Use `script/template-check.sh` to see if the template is usable.

File: CHANGELOG.md [hunk 5]
```diff
@@ -85,6 +85,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Forbid minting zero tokens. ([#1156](https://github.com/Plutonomicon/cardano-transaction-lib/issues/1156))
 - Modified functions `getWalletAddress`, `ownPubKeyHash`, `ownStakePubKeyHash`, `getWalletAddressWithNetworkTag` and `ownPaymentPubKeyHash` to return `Contract r (Array Adress)`.  ([#1045](https://github.com/Plutonomicon/cardano-transaction-lib/issues/1045))
 - `pubKeyHashAddress` and `scriptHashAddress` now both accept an optional `Credential` that corresponds to the staking component of the address ([#1060](https://github.com/Plutonomicon/cardano-transaction-lib/issues/1060))
+- `utxosAt` and `getUtxo` now use Kupo internally, `utxosAt` returns `UtxoMap` without `Maybe` context ([#1185](https://github.com/Plutonomicon/cardano-transaction-lib/pull/1185))
```
Comment (klntsky) [inline CHANGELOG.md:88]: Add a note that the config should be updated.

File: webpack.config.js [hunk 6]
```diff
@@ -23,6 +23,11 @@ module.exports = {
 
   devServer: {
     port: 4008,
+    proxy: {
+      "/matches": process.env.KUPO_HOST, // e.g. http://localhost:1442
```
Comment (klntsky) [inline webpack.config.js:27]: We should add a default value. And update the comment that says it's necessary.

**Pull request closed without merging**